### DISCUSSION TITLE
Getting 401 Unauthorized when publishing a package

### DISCUSSION THREAD

**Discussion by TheWayTC** [community/community#38782] [Packages] [answered]
### Select Topic Area

Bug

### Body

I am running into a 401 Unauthorized error when trying to push a new version of a package. I am using the same process I have used before so it's puzzling why this is happening now. I see that there is a new type of access token in Beta so maybe a bug was introduced? I am using the access token (classic) still because the new granular access token doesn't yet support managing nuget packages.

Below are the steps I used.
1. Get the new version of the package ready after updating the version number in the .csproj file.
`dotnet pack --configuration Release`
2. Push the new version of the package
`dotnet nuget push "bin/Release/TheWay.DataAccess.1.0.3.nupkg" --api-key "<access token (classic)>" --source "github"`

I double checked and my registered sources are correct. When I run `dotnet nuget list source` I get the following.
```
Registered Sources:
  1.  nuget.org [Enabled]
      https://api.nuget.org/v3/index.json
  2.  github [Enabled]
      https://nuget.pkg.github.com/TheWayInternational/index.json
```
Just to be sure I also generated a new access token (classic), but still getting the unauthorized. My access token has the following permissions: `repo, write:packages, delete:packages`. Read:packages is automatically included in write:packages.

Below is the response I get.
```
error: Unable to load the service index for source https://nuget.pkg.github.com/TheWayInternational/index.json.
error:   Response status code does not indicate success: 401 (Unauthorized).
```

**Comment by ossanna16** [comment]
Hi there @TheWayTC and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by mzros** [comment]
Any news on this?
We are facing exactly the same issue starting on 16.11.22

[very basic project](https://github.com/roXtra/NugetPush/actions)

**Reply by dariogriffo** [reply]
Have you tried filling all the information for the nuget package?

```
<PackageProjectUrl>https://github.com/MY_ORG/MY_REPO</PackageProjectUrl>
<PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
<RepositoryType>git</RepositoryType>
<RepositoryUrl>https://github.com/MY_ORG/MY_REPO</RepositoryUrl>

```

**Reply by mzros** [reply]
I added your suggestions in addition to @AdelRefaat  suggestion and now it is working.

**Comment by dariogriffo** [comment]
Same with me:
Starting from today, pushing nuget packages to my organisation's private nuget feed in Github stopped working using `${{ secrets.GITHUB_TOKEN }}`. I've also tried creating a PAT and using that instead. Still fails. I;ve been pushing packages on this repo without issues for at least 3 months.
I've updated a package version of my c# library, pushed the change and the action failed to push the nuget package.
It works locally with a PAT but won't work on actions at all.
Also other repos of my organisation works without problems, the publish.yml are identical.
I'm using `setup-dotnet@v3`
Seems someone else is having the same problem with the node registry

https://stackoverflow.com/questions/74474908/github-actions-publish-to-github-package-registry-fails-with-401-using-github-to
```

Run dotnet nuget push **/*.nupkg -s https://nuget.pkg.github.com/MY_ORG/index.json -k ***
  dotnet nuget push **/*.nupkg -s https://nuget.pkg.github.com/MY_ORG/index.json -k ***
  shell: /usr/bin/bash -e {0}
  env:
    DOTNET_ROOT: /usr/share/dotnet
error: Unable to load the service index for source https://nuget.pkg.github.com/MY_ORG/index.json.
error:   Response status code does not indicate success: [4](https://github.com/MY_ORG/MY_REPO/actions/runs/3491951967/jobs/5845188044#step:8:4)01 (Unauthorized).
```

**Comment by AdelRefaat** [comment]
I have the same issue in a private organization, it used to work with no issue, until I noticed the 401 error today 18/11/2022 
Once I removed the `/index.json` from the source url it started working

**Reply by dariogriffo** [reply]
Would you post the yaml parts with fake org and repo name? Also, what does the RepositoryUrl tag looks like in the csproj file?

**Reply by mzros** [reply]
```yaml  steps:
    - uses: actions/checkout@v3
    - name: Setup .NET
      uses: actions/setup-dotnet@v3
      with:
        dotnet-version: 7.0.x
    - name: Pack
      run: dotnet pack -o ./packed -p:RepositoryUrl=${{ github.server_url }}/${{ github.repository }} -p:PackageRequireLicenseAcceptance=false -p:RepositoryType=git
    - name: Push
      shell: pwsh
      run: dotnet nuget push ./packed/*.nupkg --source "https://nuget.pkg.github.com/${{ github.repository_owner }}" --api-key ${{ secrets.GITHUB_TOKEN }} --skip-duplicate --no-symbols```

this fixed our issue

[sample project](https://github.com/roXtra/NugetPush/actions)

**Reply by designbyadrian** [reply]
You mean today: 18/11/2022, not the 18th of December right? 😄

**Comment by kaminipandey** [comment, marked as answer]
Hi Folks, This issue is fixed now. 🎉

**Reply by TheWayTC** [reply]
Just tested my original setup and can confirm it is working again.
Thank you very much!

**Reply by VadimKhalimovskiy** [reply]
Hey! Good job!
 What was the cause of this bug?

**Comment by mdddev** [comment]
Should we revert back to appending the /index.json to the --source?

**Reply by TheWayTC** [reply]
I am using that currently and it's working fine.

**Comment by hlib-oreshchuk** [comment]
Same issue here.
